src/old-components/Guilds/common/Form/TokenAmountInput.tsx

```diff
@@ -15,33 +16,35 @@ const TokenAmountInput: React.FC<TokenAmountInputProps> = ({
   decimals = 18,
   ...rest
 }) => {
-  const [amount, setAmount] = useState<string>('');
+  const [localAmount, setLocalAmount] = useState<string>('');
+  const debouncedLocalAmount = useDebounce(localAmount, 350);
+  const localAmountBN = useStringToBigNumber(localAmount, decimals);
+
+  const valueAsString = useBigNumberToString(value, decimals);
 
-  // Set the initial value as the amount
-  const amountFromProps = useBigNumberToString(value, decimals);
   useEffect(() => {
-    if (amountFromProps != null) {
-      setAmount(amountFromProps.toString());
+    if (localAmount !== valueAsString) {
+      setLocalAmount(valueAsString);
     }
-  }, [amountFromProps]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [valueAsString]);
 
   // Call onChange when the amount changes

```

Comment:
🧹 Maybe remove this comment. I think it is self-explanatory through the code.

Reply:
Done!